File: sample-app-java+push-notification/src/main/java/com/cometchat/sampleapp/java/fcm/ui/activity/NewChatActivity.java
package com.cometchat.sampleapp.java.fcm.ui.activity;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.OnApplyWindowInsetsListener;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.cometchat.chat.core.GroupsRequest;
import com.cometchat.chat.core.UsersRequest;
import com.cometchat.chat.models.User;
import com.cometchat.chatuikit.CometChatTheme;
import com.cometchat.chatuikit.shared.interfaces.OnItemClick;
import com.cometchat.sampleapp.java.fcm.R;
import com.cometchat.sampleapp.java.fcm.databinding.ActivityNewChatBinding;
import com.google.android.material.tabs.TabLayout;
import com.google.gson.Gson;

public class NewChatActivity extends AppCompatActivity {
    private ActivityNewChatBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityNewChatBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        applyWindowInsets();
        adjustWindowSettings();

        binding.users.setToolbarVisibility(View.GONE);
        binding.groups.setToolbarVisibility(View.GONE);
        binding.users.setSeparatorVisibility(View.GONE);
        binding.groups.setSeparatorVisibility(View.GONE);
        binding.tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                if (tab.getPosition() == 0) {
                    binding.users.setVisibility(View.VISIBLE);
                    binding.groups.setVisibility(View.GONE);
                } else {
                    binding.users.setVisibility(View.GONE);
                    binding.groups.setVisibility(View.VISIBLE);
                }
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {
            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {
                // Optional logic for reselected tab
            }
        });

        binding.tabLayout.addTab(binding.tabLayout.newTab().setText(getResources().getString(R.string.app_bottom_nav_users)));
        binding.tabLayout.addTab(binding.tabLayout.newTab().setText(getResources().getString(R.string.app_bottom_nav_groups)));
        binding.tabLayout.getTabAt(0).select();
        binding.users.setUsersRequestBuilder(new UsersRequest.UsersRequestBuilder().hideBlockedUsers(true).setLimit(30));
        binding.users.setOnItemClick(new OnItemClick<User>() {
            @Override
            public void click(View view, int poUser, User user) {
                Intent intent = new Intent(NewChatActivity.this, MessagesActivity.class);
                intent.putExtra(getString(R.string.app_user), new Gson().toJson(user));
                startActivity(intent);
                finish();
            }
        });

        binding.groups.setGroupsRequestBuilder(new GroupsRequest.GroupsRequestBuilder().joinedOnly(true).setLimit(30));
        binding.groups.setOnItemClick((view, position, group) -> {
            Intent intent = new Intent(NewChatActivity.this, MessagesActivity.class);
            intent.putExtra(getString(R.string.app_group), new Gson().toJson(group));
            startActivity(intent);
            finish();
        });

        binding.ivBack.setOnClickListener(v -> finish());

        binding.tvTitle.setTextColor(CometChatTheme.getTextColorPrimary(this));
        binding.tabLayout.setTabTextColors(CometChatTheme.getTextColorSecondary(this), CometChatTheme.getPrimaryColor(this));
    }

    private void applyWindowInsets() {
        ViewCompat.setOnApplyWindowInsetsListener(binding.newChatMain, new OnApplyWindowInsetsListener() {
            @NonNull
            @Override
            public WindowInsetsCompat onApplyWindowInsets(@NonNull View v, @NonNull WindowInsetsCompat insets) {
                Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());

                v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);

                return insets;
            }
        });
    }

    private void adjustWindowSettings() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            getWindow().setDecorFitsSystemWindows(true);
        } else {
            getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);
        }
    }

    private void updateRadioButtonTextColors(ActivityNewChatBinding binding, int selectedColor, int unselectedColor) {
        // Apply the color state list to all radio buttons

    }
}

File: sample-app-kotlin+push-notification/src/main/java/com/cometchat/sampleapp/kotlin/fcm/ui/activity/NewChatActivity.kt
package com.cometchat.sampleapp.kotlin.fcm.ui.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.OnApplyWindowInsetsListener
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.cometchat.chat.core.GroupsRequest.GroupsRequestBuilder
import com.cometchat.chat.core.UsersRequest.UsersRequestBuilder
import com.cometchat.chat.models.Group
import com.cometchat.chatuikit.shared.interfaces.OnItemClick
import com.cometchat.sampleapp.kotlin.fcm.R
import com.cometchat.sampleapp.kotlin.fcm.databinding.ActivityNewChatBinding
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayout.OnTabSelectedListener
import com.google.gson.Gson

class NewChatActivity : AppCompatActivity() {
    private var binding: ActivityNewChatBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityNewChatBinding.inflate(layoutInflater)
        setContentView(binding!!.root)
        applyWindowInsets()

        binding!!.users.toolbarVisibility = View.GONE
        binding!!.groups.toolbarVisibility = View.GONE
        binding!!.users.separatorVisibility = View.GONE
        binding!!.groups.separatorVisibility = View.GONE
        binding!!.tabLayout.addOnTabSelectedListener(object : OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                if (tab.position == 0) {
                    binding!!.users.visibility = View.VISIBLE
                    binding!!.groups.visibility = View.GONE
                } else {
                    binding!!.users.visibility = View.GONE
                    binding!!.groups.visibility = View.VISIBLE
                }
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {
            }

            override fun onTabReselected(tab: TabLayout.Tab) { // Optional logic for reselected tab
            }
        })

        binding!!.tabLayout.addTab(
            binding!!.tabLayout
                .newTab()
                .setText(resources.getString(R.string.app_bottom_nav_users))
        )
        binding!!.tabLayout.addTab(
            binding!!.tabLayout
                .newTab()
                .setText(resources.getString(R.string.app_bottom_nav_groups))
        )
        binding!!.tabLayout
            .getTabAt(0)!!
            .select()
        binding!!.users.setUsersRequestBuilder(
            UsersRequestBuilder()
                .hideBlockedUsers(true)
                .setLimit(30)
        )
        binding!!.users.setOnItemClick { view, position, user ->
            val intent = Intent(this@NewChatActivity, MessagesActivity::class.java)
            intent.putExtra(getString(R.string.app_user), Gson().toJson(user))
            startActivity(intent)
            finish()
        }

        binding!!.groups.setGroupsRequestBuilder(
            GroupsRequestBuilder()
                .joinedOnly(true)
                .setLimit(30)
        )
        binding!!.groups.setOnItemClick(object : OnItemClick<Group?> {
            override fun click(
                view: View?,
                position: Int,
                group: Group?

            ) {
                val intent = Intent(this@NewChatActivity, MessagesActivity::class.java)
                intent.putExtra(getString(R.string.app_group), Gson().toJson(group))
                startActivity(intent)
                finish()
            }
        })

        binding!!.ivBack.setOnClickListener { v: View? -> finish() }
    }

    private fun applyWindowInsets() {
        ViewCompat.setOnApplyWindowInsetsListener(binding!!.newChatMain) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())

            v.setPadding(
                systemBars.left,
                systemBars.top,
                systemBars.right,
                systemBars.bottom
            )

            insets
        }
    }
}